lib/optimize/DedupePlugin.js

```diff
@@ -183,7 +183,7 @@ DedupePlugin.prototype.apply = function(compiler) {
 			source.add("(function(modules) {\n");
 			source.add(this.indent([
 				"// Check all modules for deduplicated modules",
-				"for(var i in modules) {",
+				"for(var i = 0; i < modules.length; i++) {",
```

Comment:
`modules` can be an array or an object. So better leave it `for in` but add a check for `hasOwnProperty`.